ipaserver/install/installutils.py

```diff
@@ -436,16 +436,27 @@ def format_directive(directive, value, separator, quotes, quote_char):
     fd.close()
     os.chown(filename, st.st_uid, st.st_gid) # reset perms
 
-def get_directive(filename, directive, separator=' '):
+
+def get_directive(filename, directive, separator=' ', quote_char='\"'):
     """
     A rather inefficient way to get a configuration directive.
+
+    :param filename: input filename
+    :param directive: directive name
+    :param separator: separator between directive and value
+    :param quote_char: the characters that are used in this particular config
+        file to quote values. This character will be stripped and unescaped
+        from the raw value.
+
+    :returns: The (unquoted) value if the directive was found, None otherwise
     """
     fd = open(filename, "r")
     for line in fd:
         if line.lstrip().startswith(directive):
             line = line.strip()
             result = line.split(separator, 1)[1]
-            result = result.strip('"')
+            result = result.strip(quote_char)
```

Comment:
I think that this usage of `strip` will cause problems if the text in config file is like `\"abc\"`. It will remove only the last `"` (which is incorrect anyway because `"` are part of the string and not quotation).